dot/sync/state_sync.go

```diff
@@ -0,0 +1,209 @@
+package sync
+
+import (
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/network/messages"
+	"github.com/ChainSafe/gossamer/dot/peerset"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/ChainSafe/gossamer/pkg/trie"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+type StateStorage interface {
+	StoreTrie(ts *storage.TrieState, header *types.Header) error
+}
+
+type StateSyncStrategy struct {
+	// Strategy dependencies and config
+	peers                *peerViewSet
+	badBlocks            []string
+	reqMaker             network.RequestMaker
+	blockState           BlockState
+	storage              StateStorage
+	stateRequestProvider *StateRequestProvider
+
+	// State sync state
+	startedAt   time.Time
+	targetBlock types.Header
+	completed   bool
+}
+
+type StateSyncStrategyConfig struct {
+	Telemetry    Telemetry
+	BadBlocks    []string
+	BlockState   BlockState
+	Peers        *peerViewSet
+	ReqMaker     network.RequestMaker
+	TargetBlock  types.Header
+	StateStorage StateStorage
+}
+
+func NewStateSyncStrategy(
+	cfg *StateSyncStrategyConfig,
+) *StateSyncStrategy {
+	return &StateSyncStrategy{
+		peers:       cfg.Peers,
+		badBlocks:   cfg.BadBlocks,
+		blockState:  cfg.BlockState,
+		targetBlock: cfg.TargetBlock,
+		reqMaker:    cfg.ReqMaker,
+		storage:     cfg.StateStorage,
+		// TODO: we can assume that v1 is right for every chain but we need to find a way to set the right state version
```

Comment:
That's is a good point! Actually we should query the information from the runtime, e.g is paseo using v1?

Reply:
Yeah, that's fair, I would suggest to keep the TODO as an issue like: "retrieve state version from runtime at state sync". 

> But how can we query it if we don't have the state yet?

Good question, can you place this question together with the TODO comment and in the issue so we keep track of it.